src/webgl/3d_primitives.js

```diff
@@ -520,6 +520,94 @@ function primitives3D(p5, fn){
     return this._renderer.endGeometry();
   };
 
+
+/**
+ * Sets the stroke rendering mode to balance performance and visual features when drawing lines.
+ *
+ * `strokeMode()` offers two modes:
+ *
+ * - `SIMPLE`: Optimizes for speed by disabling caps, joins, and stroke color features.
+ *   Use this mode for faster line rendering when these visual details are unnecessary.
+ * - `FULLY`: Enables caps, joins, and stroke color for lines.
+ *   This mode provides enhanced visuals but may reduce performance due to additional processing.
+ *
+ * Choose the mode that best suits your application's needs to either improve rendering speed or enhance visual quality.
+ *
+ * @method strokeMode
+ * @param {string} mode - The stroke mode to set. Possible values are:
+ *   - `'SIMPLE'`: Fast rendering without caps, joins, or stroke color.
+ *   - `'FULLY'`: Detailed rendering with caps, joins, and stroke color.
+ *
+ * @example
+ * <div>
+ * <code>
+ * function setup() {
+ *   createCanvas(100, 100, WEBGL);
+ *   strokeWeight(10);
+ *   stroke(0);
+ * 
+ *  describe('A red, horizontal, rectangular shape with rounded edges (resembling a pill or capsule) in a grey background');
+ * }
+ * 
+ * function draw() {
+ *   strokeMode(FULLY); // Enables detailed rendering with caps, joins, and stroke color.
+ *   stroke('red');
+ * 
+ *   background(128);
+ *   
+ *   let centerX = 0;
+ *   let centerY = 0;
+ *
+ *   // Length of the small centered line
+ *   let lineLength = 50; 
+ * 
+ *   beginShape(LINES);
+ *   vertex(centerX - lineLength / 2, centerY, 0);
```

Comment:
How do you feel about having a 3D shape like a default `sphere()`, and maybe a single Bezier curve in each of these examples? I feel like those are two cases where the difference is clear: the sphere probably looks the same regardless, and the curve (especially with a higher stroke weight) will look very disjoint without full curve decorations.

We can also make the examples more than 100x100 if that makes it easier to see since I think the new site's layout accommodates that decently.